### ISSUE TITLE
add a guide on how to compile all this emscripten

### ISSUE THREAD

**Issue by MyTDT-Mysoft** [emscripten-core/emscripten#14259]
because since you guys dont provide 32bit binaries... at least add an entry on how to compile for 32bit... altough i see no reason to not have 32bit releases except laziness...

**Comment by sbc100** [comment]
So many lazy folks around here!

**Comment by sbc100** [comment]
In all seriousness, it is quite a lot of work for many different folks to add support building release binaries for new platforms.   For example, we recently invested quite a bit in building arm64 binaries for macOS since there have been a lot of requests for that target recently and more folks are moving to  arm64 versions of macOS.   Conversely, the number of folks running 32-bit windows and 32-bit linux, especially those who are also developers, seems to be very small and getting smaller.   I think its been a few years now since anyone has requested a 32-bit SDK.

The amount of work required to support 32-bit builds is simply not worth it given that tiny number of potential users.  The incremental cost of adding new platforms is not just in the codebase itself, but also in the CI systems that build the binaries, the CPU cycles and storage used by such systems.

For similar reasons we choose just 3 primary platforms to actually run our tests on (node, chrome, firefox).  Continuously testing on additional platforms either at runtime or at build time comes with a high cost.

None of this is to say that supporting lesser used platforms is undesirable, or that we would accept patches to support them, its most a matter of working with very limited resources.

**Comment by MyTDT-Mysoft** [comment]
yes, i understand hence why, altough on windows supporting 32bit shouldnt be hard, its probabily mainly to thirdy party stuff or things deliberated avoiding to support that, but either way, thats why i asked on a gui on how to build that stuff... like what emsdk build, and how... or how it would be for someone compiling for other platforms... because its right now a too different environment than i'm used so, some guidance on what would be necessary, and the order or alternative would be trully welcome :)

but i have to say something... you guys have all this tools to make it easy to support platforms, but on the other hand, less and less effort is done to support other platforms, solely based on popularity, so you have the tools to make it easier... you use python because it would suppose to be easier... but from what i see all those tools are making it HARDER to use on other platforms not easier... so that to me sounds like a lack of planning... or too early adoption of "popular" stuff that are nothing but lies... (i'm more a person that tend to priorize groups that VALUE most... not  groups that are biggest or more popular, as those are usually the ones that just want to use&abuse)

**Comment by sbc100** [comment]
Did you see the have instructions we have on how to build emscripten's dependencies from source:  https://emscripten.org/docs/building_from_source/index.html.   They should be fairly platform neutral instructions.  If you have ideas for how to improve them those would be welcome.

I don't what the problem is with using python.. doesn't python work find on win32?    You can download and install it from python.org can't you?

**Comment by kripken** [comment]
@MyTDT-Mysoft We have also provided you with some 32-bit binaryen builds here, have you tested them? https://github.com/WebAssembly/binaryen/issues/3903#issuecomment-846599069
